nat-gateway.tf

```diff
@@ -15,7 +15,7 @@ locals {
 }
 
 resource "aws_eip" "default" {
-  count = local.enabled ? local.nat_gateway_eip_count : 0
+  count = local.public_enabled ? local.nat_gateway_eip_count : 0
```

Comment:
This raises the question of whether or not we should even allow this module to create only private subnets, since they would have no means of egress. 

In any case, this is wrong, because we only can and need to create a NAT gateway or instance if we have _both_ public and private subnets.

Reply:
yes, NAT gets created in public subnets.
Without NAT, the private subnets can't communicate with anything (and many applications will not work b/c of that).
@alexjurkiewicz what's the use case for having only private (and completely isolated from everything) subnets?